[5.3] build.php Bug fix and stability improvements

muhme opened the pull request:
### Summary of Changes

This is a follow-up to https://github.com/joomla/joomla-cms/pull/45535 and was discussed in maintainer meeting 11 June 2025.

- Copying of non-existing `build/fido.jwt` file is simple deleted
  - I was wondering, why do we have this error and nobody cares? Looking around I found:
    - `plugins/system/webauthn/fido.jwt` file is created by `composer i`
      - Which runs `php build/update_fido_cache.php`
        - Which downloads the FIDO metadata about authenticators from https://mds.fidoalliance.org into `plugins/system/webauthn/fido.jwt` file
    - The line with `system('cp build/fido.jwt ' . $fullpath . '/plugins/system/webauthn/fido.jwt');` was added on 2 Sep 2022 with c2191a7 'Joomla! 4.2.2 Stable'
    - I assume this was another implementation of getting the FIDO metadata and is no more valid.
    - btw, the check `file_exists 'plugins/system/webauthn/fido.jwt'` was added with 205c7a5 'Fix/38233 webauth preload fido (#38664)'
- Improvement of two places where a variable was used for both the return code and the system output
- The exit codes are now checked for all system commands and the script stops in case of errors
- Before this PR, the zip commands displayed every added file. But before [#45535](https://github.com/joomla/joomla-cms/pull/45535), the output of the zip command with all added file names was suppressed by redirecting stdout to `/dev/null`. The same behaviour is now restored with the -q option.

### Testing Instructions

1. Run `php build/build.php` 
2. Run `php build/build.php --remote=HEAD`
3. Choose one `system_and_check` call and *hack* an error in, e.g. change `composer` to  `composerXXX` and check the script stops
4. *Hack* one `passthru()` replacement e.g. change `which git` with `which cheesecake` and check the script stops

check the latest tag from Git repository is used, e.g.
 `Start build for remote tags/6.0.0-alpha1.`
  
After merging this PR, we should check the nightly builds as they also use the script.

### Actual result BEFORE applying this Pull Request

* Error message line
  `cp: build/fido.jwt: No such file or directory`
* Release build is working
* Output is ~30'000 lines

### Expected result AFTER applying this Pull Request

* No error message line
* Release build is working
* Output is ~5'000 lines

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [X] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [X] No documentation changes for manual.joomla.org needed

muhme committed:
Bug fix and stability improvements

- Copying of non-existing `build/fido.jwt` file is simple deleted
  - I was wondering, why do we have this error and nobody cares? Looking around I found:
    - `plugins/system/webauthn/fido.jwt` file is created by `composer i`
      - Which runs `php build/update_fido_cache.php`
        - Which downloads the FIDO metadata about authenticators from
          https://mds.fidoalliance.org into `plugins/system/webauthn/fido.jwt` file
    - The line with `system('cp build/fido.jwt ' . $fullpath . '/plugins/system/webauthn/fido.jwt');`
      was added on 2 Sep 2022 with c2191a7 'Joomla! 4.2.2 Stable'
    - I assume this was another implementation of getting the FIDO metadata and is no more valid.
    - btw, the check `file_exists 'plugins/system/webauthn/fido.jwt'` was added with 205c7a5
      'Fix/38233 webauth preload fido (#38664)'
- Improvement of two places where a variable was used for both the return code and the system output
- The exit codes are now checked for all system commands and the script stops in case of errors

muhme committed:
php-cs-fixer

richard67 commented on build/build.php:
```diff
@@ -224,32 +223,59 @@ function clean_composer(string $dir)
 $time = time();
 
 // Set path to git binary (e.g., /usr/local/git/bin/git or /usr/bin/git)
 ob_start();
```
Do we sill need the `ob_start();` here when replacing the `ob_get_clean()` with `capture_or_fail` (which doesn't include any `ob_` operations) a few lines below this?

muhme replied:
:+1: In deed it it true, `ob_start()` is no longer needed. Deleted. thx

muhme committed:
Fix Review Finding

In deed it it true, ob_start is no longer needed. thx

muhme committed:
Suppress the logging of all file names in zip

Before this commit, the zip commands displayed every added file.
Before #45535, the output of the zip command with all added file names
was suppressed by redirecting stdout to /dev/null.
The same behaviour is now restored with the -q option.

muhme commented:
@tecpromotion May I ask you to test with macOS?
@Bodge-IT @softforge Could one of you do a test with Windows?

tecpromotion commented:
I have tested this item :white_check_mark: successfully on ef8e99fc17ba7f6da7529460430370232c4a74f7<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/45622">issues.joomla.org/tracker/joomla-cms/45622</a>.</sub>

richard67 committed:
Merge branch '5.3-dev' into 53-build-php

richard67 commented:
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/45622">issues.joomla.org/tracker/joomla-cms/45622</a>.</sub>

rdeutz commented:
Thanks
